apps/api/src/campaign/campaign.service.ts

```diff
@@ -1037,9 +1037,17 @@ export class CampaignService {
 
   async createCampaignNotificationList(updated: { title: string; id: string }) {
     // Generate list in the marketing platform
-    const listId = await this.marketingNotificationsService.provider.createNewContactList({
-      name: updated.title || updated.id,
-    })
+    let listId: string
+    const lists = await this.marketingNotificationsService.provider.getContactLists()
+    const campaginEmailLists = lists.body.result
+    const exists = campaginEmailLists.find((campaign) => campaign.name === updated.title)
+    if (exists) {
+      listId = exists.id
+    } else {
+      listId = await this.marketingNotificationsService.provider.createNewContactList({
```

Comment:
do we want to silently create a list here? I would assume we should be only doing anything if a list already exists?

Reply:
The current issue is that for some reason, email lists for some campaigns exist in Sendgrid database, but don't exist in ours NotificationList. This causes an attempt to create two email lists with the same name, which results in Bad Request whenever somebody attempts to subscribe for email notifications to particular campaign.

What this peace of code does is to check, whether there is already a Sendgrid list with the same name, before attempting to create a new one.